### DISCUSSION TITLE
How can I set up the Selector so that the Style only applies to itself?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#14134] [Q&A] [answered]
For example:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication4.MainWindow"
		xmlns:Local="using:AvaloniaApplication4"
        Title="AvaloniaApplication4">
	<Border>
		<Border.Styles>
			<Style Selector="Border">
				<Setter Property="Height" Value="300"></Setter>
				<Setter Property="Width" Value="300"></Setter>
			</Style>
		</Border.Styles>
		<Border Background="Red">			
		</Border>
	</Border>	
</Window>
```

As you see, the style will apply to all of its descendants.
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/caa66094-8fb4-4ca7-b868-00687289528e)

In spite I can use the selector by name, for example:
`<Style Selector="#myButton">`

However, I cannot guarantee that my users will use the same name in my usercontrol.

Is there any way that I can set up the Selector only work for itself without given a name?

**Comment by timunie** [comment, marked as answer]
In that case inherit. MyButton: Button 

With that, style MyButton and the user can simply use <MyButton />

**Reply by kyuranger** [reply]
Thanks a lot.
